packages/blade/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,524 @@
+  }
+
+  // A special case where we hide the indicators when the carousel is responsive
+  // Because indicators become useless since it's not aparent which carousel item is active
+  // and how many carousel items are visible at a time
+  if (isResponsive && !shouldAddStartEndSpacing && !isMobile) {
+    showIndicators = false;
+  }
+
+  const isNavButtonsOnSide = !isResponsive && navigationButtonPosition === 'side';
+  const shouldNavButtonsFloat = isResponsive && navigationButtonPosition === 'side';
+  const totalNumberOfSlides = React.Children.count(children);
+  const numberOfIndicators = Math.ceil(totalNumberOfSlides / _visibleItems);
+
+  // hide next/prev button on reaching start/end when carousel is responsive
+  // in non-responsive carousel we always show the next/prev buttons to allow looping
+  const shouldShowPrevButton = isResponsive ? activeSlide !== 0 : true;
+  const shouldShowNextButton = isResponsive ? activeSlide !== totalNumberOfSlides - 1 : true;
+
+  // calculate the start/end margin so that we can
+  // deduct that margin when scrolling to a carousel item with goToSlideIndex
+  React.useLayoutEffect(() => {
+    // Do not calculate if not needed
+    if (!isResponsive && !shouldAddStartEndSpacing) return;
+    if (!containerRef.current) return;
+
+    const carouselItemId = getCarouselItemId(id, 0);
+    const carouselItem = containerRef.current.querySelector(carouselItemId);
+    if (!carouselItem) return;
+
+    const carouselItemLeft = carouselItem.getBoundingClientRect().left ?? 0;
+    const carouselContainerLeft = containerRef.current.getBoundingClientRect().left ?? 0;
+
+    setStartEndMargin(carouselItemLeft - carouselContainerLeft);
+  }, [id, isResponsive, shouldAddStartEndSpacing]);
+
+  const goToSlideIndex = (slideIndex: number) => {
+    if (!containerRef.current) return;
+
+    const carouselItemId = getCarouselItemId(id, slideIndex * _visibleItems);
+    const carouselItem = containerRef.current.querySelector(carouselItemId);
+    if (!carouselItem) return;
+
+    const carouselItemLeft =
+      carouselItem.getBoundingClientRect().left -
+        containerRef.current.getBoundingClientRect().left ?? 0;
+    const left = containerRef.current.scrollLeft + carouselItemLeft;
+
+    containerRef.current.scroll({
+      left: left - startEndMargin,
+      behavior: 'smooth',
+    });
+    setActiveSlide(slideIndex);
+    setActiveIndicator(slideIndex);
+  };
+
+  const goToNextSlide = () => {
+    let slideIndex = activeSlide + 1;
+    if (slideIndex >= numberOfIndicators) {
+      slideIndex = 0;
+    }
+
+    // an edge case where if carousel is responsive
+    // and shouldHaveStartEndSpacing is set to false
+    // there can be a case where numberOfIndicators is set to 10 but
+    // visually there is 3 or 4 items, in those cases we want to check if we reached the
+    // end of the scroll container if so we wrap around
+    // TODO: we should probably hide the indicators in this case
```

Comment:
Addressing this?

Reply:
Yes addressed. removed the TODO, even when indicators are hidden humko wrap around to karna padega, so this code stays.